src/lib/features/change-request-segment-usage-read-model.ts/sql-change-request-segment-usage-read-model.ts

```diff
@@ -0,0 +1,33 @@
+import { Db } from '../../db/db';
+import { IChangeRequestSegmentUsageReadModel } from './change-request-segment-usage-read-model';
+
+export class ChangeRequestSegmentUsageReadModel
+    implements IChangeRequestSegmentUsageReadModel
+{
+    private db: Db;
+
+    constructor(db: Db) {
+        this.db = db;
+    }
+
+    public async isSegmentUsedInActiveChangeRequests(
+        segmentId: number,
+    ): Promise<boolean> {
+        const result = await this.db.raw(
+            `SELECT *
+            FROM change_request_events
+            WHERE change_request_id IN (
+            SELECT id
```

Comment:
Generally good to avoid nested queries due to bad performance. Use joins instead.

Reply:
Done in 430bb2b77e893081eff3b6f02b48d16159866619